esf.tf

```diff
@@ -33,10 +37,11 @@ data "external" "esf_lambda_loader" {
 }
 
 module "esf-lambda-function" {
+  for_each = toset(data.aws_sqs_queue.esf-queue)
```

Comment:
it seems we are creating a lambda function for each `data.aws_sqs_queue.esf-queue`, is it correct?
that's not needed/wanted: still one single lambda, only `aws_lambda_event_source_mapping` has to be for each of the above.

Reply:
also instead of `data.aws_sqs_queue.esf-queue` we can directly use `var.sqs_esf_names`